src/python/pants/engine/internals/graph.py

```diff
@@ -992,30 +922,33 @@ async def resolve_unparsed_address_inputs(
 class NoApplicableTargetsException(Exception):
     def __init__(
         self,
-        targets_with_origins: TargetsWithOrigins,
+        targets: Targets,
         *,
         applicable_target_types: Iterable[Type[Target]],
         goal_description: str,
     ) -> None:
         applicable_target_aliases = sorted(
             {target_type.alias for target_type in applicable_target_types}
         )
-        inapplicable_target_aliases = sorted({tgt.alias for tgt in targets_with_origins.targets})
-        specs = sorted(
-            {str(target_with_origin.origin) for target_with_origin in targets_with_origins}
-        )
```

Comment:
So, this code used to group by origin in order to report whether one of the origins did not match anything, and that was one of the reasons for the Origin code (see #8402, where it was called "provenanced").

If we're ok losing that functionality, then this looks fine. But as mentioned in Slack: if _only_ this code path (`find_valid_field_sets_for_target_roots`) needs to consume origins, and all other consumers want to strip the Origin information, is there a middle ground where this API requests `TargetsWithOrigins`, but then `TargetRootsToFieldSets` just contains `Targets` rather than `TargetsWithOrigins`?

Reply:
>  this code used to group by origin in order to report whether one of the origins did not match anything, 

Do you mean this error message did the grouping, or somewhere else? This error message simply echoes back all the specs. Note that this error only happens if none of the specs are valid, meaning there is no grouping - they're all bad.

> is there a middle ground where this API requests TargetsWithOrigins

Yes, and I tried this approach at first, but it means we can't delete all this PR's code. Given how little we use `WithOrigins`, and how much complexity we added in 2.0 with file targets, I think this simplification is quite valuable.

Reply:
> Do you mean this error message did the grouping, or somewhere else?

Yes, there were a few spots in this file that were grouping by origin, including the line I commented on.

> Given how little we use WithOrigins, and how much complexity we added in 2.0 with file targets, I think this simplification is quite valuable.

Agreed. If @benjyw is fine with getting rid of the #8402 functionality (or if we think it already wasn't working), then I'm definitely fine with it.

Reply:
> including the line I commented on.

Hm, the commented lines weren't doing any grouping. They only sort the original input specs, all of them. If we wanted to do that, we could request `Specs` in the original rule, and output that here in the error.

> Agreed. If @benjyw is fine with getting rid of the #8402 functionality

Cool. I talked to him yesterday and he was okay with the change.

Reply:
Does this mean that 

./pants test not/a/test/target

Will now silently no-op?

Reply:
It's done that for the past few months. https://github.com/pantsbuild/pants/pull/11051 speaks to why that behavior is really important: `./pants --changed-since`. If we errored on unmatched literal addresses like before, then `./pants --changed-since test` would break if you had any non test files in the mix.

We could add a warning if nothing is matched though, rather than a silent no-op. Some goals like `py-constraints` and `list` warn.

Reply:
SGTM